yapapi/payload/package.py

```diff
@@ -28,11 +33,64 @@ async def resolve_package_url(image_url: str, image_hash: str) -> str:
         return f"hash:sha3:{image_hash}:{image_url}"
 
 
-async def resolve_package_repo_url(repo_url: str, image_hash: str) -> str:
+def sizeof_fmt(num, suffix="B"):
+    for unit in ["", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "Zi"]:
+        if abs(num) < 1024.0:
+            return f"{num:3.2f}{unit}{suffix}"
+        num /= 1024.0
+    return f"{num:.1f}Yi{suffix}"
+
+
+async def resolve_package_url(
+    repo_url: str,
+    image_tag: Optional[str] = None,
+    image_hash: Optional[str] = None,
+    image_use_https: bool = False,
+) -> str:
     async with aiohttp.ClientSession() as client:
-        resp = await client.get(f"{repo_url}/image.{image_hash}.link")
+        is_dev = os.getenv("GOLEM_DEV_MODE", False)
+
+        if image_tag is None and image_hash is None:
+            raise PackageException("Neither image tag nor image hash specified")
+
+        if image_tag and image_hash:
+            raise PackageException("Both image tag and image hash specified")
+
+        if image_tag:
+            url_params = f"tag={image_tag}"
+        else:
+            url_params = f"hash={image_hash}"
+
+        if is_dev:
+            # if dev, skip usage statistics, pass dev option for statistics
+            url_params += "&dev=true"
+        else:
+            # resolved by yapapi, so count as used tag (for usage statistics)
+            url_params += "&count=true"
+
+        query_url = f"{repo_url}/v1/image/info?{url_params}"
```

Comment:
there's no point in building a http query string like that -> it's much better handled by appropriate arguments to the `client.get` call below